zh-learn-infrastructure/src/main/java/com/zhlearn/infrastructure/common/ExampleResponseMapper.java

```diff
@@ -65,10 +64,14 @@ public Example apply(String yamlResponse) {
             
             return new Example(allUsages, seriesItems);
             
-        } catch (IOException e) {
-            log.error("Failed to parse YAML response: {}", e.getMessage(), e);
+        } catch (IllegalStateException e) {
```

Comment:
remove this catch and rethrow, and remove the catch Exception below. If there are Checked Exceptions, catch ONLY those AND rethrow them wrapped in an Unchecked exception